src/views/Teams/TeamList.jsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import { Link } from 'react-router-dom';
+import { deleteTeamById, getTeams } from '../../services/teams';
+
+function TeamList() {
+  const [teams, setTeams] = useState([]);
+  const history = useHistory();
+  const [loading, setLoading] = useState(true);
+
+  const loadTeams = async () => {
+    setLoading(true);
+    const response = await getTeams();
+    setTeams(response);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    loadTeams();
+  }, []);
+
+  const handleUpdate = (id) => {
+    history.push(`/teams/update/${id}`);
+  };
+
+  const handleDelete = async ({ id, name }) => {
```

Comment:
If the user tries to delete a team with any players associated with it, it fails and doesn't give a useful error. You can fix this by giving the user an error that tells them to delete all players on the team first or adding the player deletion step to the team delete handler here. Either will work